invokeai/frontend/web/src/services/api/endpoints/boardImages.ts

```diff
@@ -85,6 +96,60 @@ export const boardImagesApi = api.injectEndpoints({
         }
       },
     }),
+    addMultipleImagesToBoard: build.mutation<void, AddImagesToBoardArg>({
+      query: ({ board_id, image_names }) => ({
+        url: `board_images/${board_id}/add_images`,
+        method: 'POST',
+        body: image_names,
+      }),
+      invalidatesTags: (result, error, arg) => [
+        { type: 'Board', id: arg.board_id },
+      ],
+      async onQueryStarted(
+        { image_names, ...patch },
+        { dispatch, queryFulfilled }
+      ) {
+        const patchResults: PatchCollection[] = [];
+
+        image_names.forEach(image_name => {
+          patchResults.push(dispatch(
+            imagesApi.util.updateQueryData('getImageDTO', image_name, (draft) => {
+              Object.assign(draft, patch);
+            })
+          ));
+        })
+        try {
+          await queryFulfilled;
+        } catch {
+          patchResults.forEach(patch => patch.undo());
+        }
+      },
+    }),
+
+    removeMultipleImagesFromBoard: build.mutation<void, RemoveImagesFromBoardArg>({
+      query: ({ board_id, image_names }) => ({
+        url: `board_images/${board_id}/remove_images`,
+        method: 'POST',
+        body: image_names,
+      }),
+      invalidatesTags: (result, error, arg) => [
+        { type: 'Board', id: arg.board_id },
+      ],
+      async onQueryStarted(
+        { image_names, ...patch },
+        { dispatch, queryFulfilled }
+      ) {
+        const patchResults: PatchCollection[] = [];
+
+        image_names.forEach(image_name => {
+          patchResults.push(dispatch(
+            imagesApi.util.updateQueryData('getImageDTO', image_name, (draft) => {
+              Object.assign(draft, { board_id: null });
+            })
+          ));
```

Comment:
I don't think this will still be needed once we are not filtering the main image adapter by board_id but left it for now